src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,150 @@
+        return _download_binary()
+
+
+# Returns path to existing draftv2 binary and None otherwise
+def _get_existing_path() -> str:
+    print('Checking if DraftV2 binary exists locally...')
+
+    operating_system = platform.system().lower()
+    # Filename depends on the operating system
+    filename = 'draftv2-' + operating_system + '-amd64'
+
+    paths = _get_potential_paths()
+    if not paths:
+        print('List of potential DraftV2 paths is empty')
+        return None
+
+    for path in paths:
+        binary_file_path = path + '/' + filename
+        if os.path.exists(binary_file_path):
+            print('Existing binary found at: ' + binary_file_path)
+            return binary_file_path
+    return None
+
+
+# Returns a list of potential draftV2 binary paths
+def _get_potential_paths() -> List[str]:
+    result = [str(Path.home()) + '/' + '.aksapp']
+    paths = os.environ['PATH'].split(':')
+
+    for path in paths:
+        if 'draftv2' in path:
+            result.append(path)
+    return result
+
+
+def _build_arguments(app_name: str,
+                     language: str,
+                     create_config: str,
+                     dockerfile_only: str,
+                     deployment_only: str) -> List[str]:
+    options = {
+        'app-name': app_name,
+        'language': language,
+        'create-config': create_config,
+        'dockerfile-only': dockerfile_only,
+        'deployment-only': deployment_only
+    }
+    args_list = []
+    for arg, val in options.items():
+        if val:
+            args_list.append(f'--{arg}={val}')
+    return args_list
+
+
+def _run(binary_path: str, deployment_path: str, arguments: List[str]) -> bool:
+    if binary_path is None:
+        raise ValueError('The given Binary path was null or empty')
+
+    print('Running DraftV2 Binary ...')
+    cmd = [binary_path, 'create', deployment_path] + arguments
+    process = subprocess.Popen(cmd)
+    exit_code = process.wait()
+    return exit_code == 0
+
+
+def _cmd_finish():
+    # Clean up logic can go here if needed
+    pass
+
+
+def _download_binary() -> str:
+    # prompt user to download binary. If users says no, we error out and tell them that this requires the binary
+    from knack.prompting import prompt_y_n
+    msg = 'The required binary was not found. Would you like us to download the required binary for you?'
+
+    if not prompt_y_n(msg, default='n'):
+        raise ValueError('`az aks app` requires the missing dependency')
+
+    print('Attempting to download dependency...')
+
+    operating_system = platform.system()
+    draftv2_release_version = 'v0.0.5'
+    filename = 'draftv2-' + operating_system.lower() + '-amd64'
+    url = 'https://github.com/Azure/aks-app/releases/download/' + draftv2_release_version + '/' + filename
+    headers = {'Accept': 'application/octet-stream'}
+
+    dir_name = '.aksapp'
+    # Downloading the file by sending the request to the URL
+    req = requests.get(url, headers=headers)
```

Comment:
💡 We could get inputs form existing `cli` devs to double check if they have any other preferred way for `download`